### DISCUSSION TITLE
Config file for Lilygo T-HMI

### DISCUSSION THREAD

**Discussion by WhyDoYouMakeUsDoThis** [HASwitchPlate/openHASP#477] [Hardware]
Hi, 
  The device is listed as a suggested under the 0.7.0-rc4 devices.  It's listed as under development..   Does anyone have a .ini file for compiling it for this device?
I've tried basing it off a couple ESP32S3 with parallel ST7789V.  I can get the system to boot as I can see it at the standard first boot in the console, but no display... So I have something(maybe many things) wrong.
I was wondering if anyone had a working config.

Thanks

**Comment by WhyDoYouMakeUsDoThis** [comment]
Here's the link to the product page:
https://www.lilygo.cc/en-ca/products/t-hmi?_pos=1&_psq=T-HM&_ss=e&_v=1.0

Edited the attempt away as the include one works perfectly.

**Reply by fvanroie** [reply]
I added `user_setups/esp32s3/lilygo-t-hmi.ini` to the master repository. Feel free to pull the latest changes from GitHub and try it out.

**Comment by WhyDoYouMakeUsDoThis** [comment]
Thanks so much!  That config works great so far.    I need to rebuild my pages for a smaller screen and start testing the device.

**Reply by fvanroie** [reply]
Great. Please report back if you have any issues with this device.

**Reply by WhyDoYouMakeUsDoThis** [reply]
There seems to be issues with the backlight.   I can't set the pin in the GUI, 38 isn't an option although it is defined in the profile.  There are times on boot the display is fully blank, I can access via the web or serial, just no display. 
It never turns off if on... Setting to factory seems to bring it back.
I'd suspect it's all related to the BL GPIO.   Just a wild guess.
My board is stamped with version 1.2. So it might still be in flux

Fully willing to provide logs or test specifc things if needed.

**Reply by WhyDoYouMakeUsDoThis** [reply]
Backlight functions once connected to HA.  (I used an older 0.6.3 version to get in to HA, then renamed the 0.7 version to match)

**Comment by WhyDoYouMakeUsDoThis** [comment]
One other thing about the platform.  The WiFi antenna on-board isn't very good.   Mine is sitting 3m direct line of sight to my base station.  It ranged from -83 dBm all the way up to -73 dBm.  Both terrible.
It has an IPEX MHF4 connector.  With a 8dBi antenna attached the unit gets a -43dBm.
Just for people to be aware of.
